src/trials.jl

```diff
@@ -338,115 +346,116 @@ Base.show(io::IO, t::TrialEstimate) = _show(io, t)
 
     pad = get(io, :pad, "")
-    print(io, "BenchmarkTools.Trial: ", length(t), " sample", if length(t) > 1 "s" else "" end,
-          " with ", t.params.evals, " evaluation", if t.params.evals > 1 "s" else "" end ,".\n")
+    padcolor = :light_black
+
+    showpercentile = _percentile()
 
     perm = sortperm(t.times)
     times = t.times[perm]
     gctimes = t.gctimes[perm]
 
-    if length(t) > 1
-        med = median(t)
-        avg = mean(t)
-        std = Statistics.std(t)
-        min = minimum(t)
-        max = maximum(t)
-
-        medtime, medgc = prettytime(time(med)), prettypercent(gcratio(med))
-        avgtime, avggc = prettytime(time(avg)), prettypercent(gcratio(avg))
-        stdtime, stdgc = prettytime(time(std)), prettypercent(Statistics.std(gctimes ./ times))
-        mintime, mingc = prettytime(time(min)), prettypercent(gcratio(min))
-        maxtime, maxgc = prettytime(time(max)), prettypercent(gcratio(max))
-
-        memorystr = string(prettymemory(memory(min)))
-        allocsstr = string(allocs(min))
+    if length(t) == 0
+        print(io, "BenchmarkTools.Trial: 0 samples")
+        return
     elseif length(t) == 1
-        print(io, pad, " Single result which took ")
-        printstyled(io, prettytime(times[1]); color=:blue)
-        print(io, " (", prettypercent(gctimes[1]/times[1]), " GC) ")
-        print(io, "to evaluate,\n")
-        print(io, pad, " with a memory estimate of ")
-        printstyled(io, prettymemory(t.memory[1]); color=:yellow)
-        print(io, ", over ")
-        printstyled(io, t.allocs[1]; color=:yellow)
-        print(io, " allocations.")
+        println(io, "BenchmarkTools.Trial:")
+        # Time
+        print(io, pad, "│  Only 1 sample: ")
+        printstyled(io, prettytime(times[1]); color=:green)
+
+        # Memory
+        println(io)
+        print(io, pad, "│  ", prettycount(t.allocs[1]), " allocation", t.allocs[1]==1 ? "" : "s")
+        if t.allocs[1] > 0
+            print(io, ", ", prettymemory(t.memory[1]))
+        end
+
+        # GC time
+        if t.gctimes[1] > 0
+            println(io)
+            print(io, pad, "│  GC time: ", prettytime(t.gctimes[1]))
+            printstyled(io, " (", prettypercent(t.gctimes[1] / t.times[1]),")"; color=:green)
+        end
         return
+    end # done with trivial cases.
+
+    med = median(t)
+    avg = mean(t)
+    min = minimum(t)
+    max = maximum(t)
+    q99 = quantile(t, showpercentile/100)
+
+    mintime = prettytime(time(min))
+    medtime = prettytime(time(med))
+    avgtime = prettytime(time(avg))
+    q99time = prettytime(time(q99))
+
+    # Mean GC time is just that; then we take the percentage of the mean time
+    avggctime, avegcpercent = prettytime(mean(gctimes)), prettypercent(mean(gctimes) / mean(times))
+    q99gctime = prettytime(quantile(gctimes, showpercentile/100))
+    # Maximum GC time has a percentage which is of the same run, not necc. the longest run
+    _t, _i = findmax(gctimes)
```

Comment:
This is the %(max gc), which may not be the same as the max(% gc). I think you're computing the quantity we want, but clarify this in the comments. (The display is fairly clear.)

Reply:
Yes, my intention is to find the maximum GC time, and then present the percentage this makes up of that same run. Which may not be the slowest run, in pathalogical cases. 

My example is this, which I would like to summarise this as `GC time: mean 526 μs (3.85%), max 9.914 ms (75.99%)`
```
julia> b = @benchmark (rand()<0.3 ? sleep(0.03) : sum(rand(10^6)))
BenchmarkTools.Trial: 367 samples with 1 evaluation.
 Range (min … max):   1.057 ms … 36.199 ms  ┊ GC (min … max): 0.00% …  0.00%
 Time  (median):      5.123 ms              ┊ GC (median):    0.00%
 Time  (mean ± σ):   13.686 ms ± 14.512 ms  ┊ GC (mean ± σ):  3.85% ± 18.38%

julia> findmax(b.gctimes)
(9.914125e6, 56)

julia> ans[1] / b.times[ans[2]]
0.75992632624788

julia> mean(b.gctimes)
526577.2070844687

julia> ans / mean(b.times)  # is displayed above, and stays
0.03847665760028995

julia> std(b.gctimes ./ b.times)  # is displayed above, but removed by this PR
0.1837843069503372

julia> maximum(b.gctimes ./ b.times)  # in this example, max(percentage) agrees with maximum(gc) as a percentage
0.75992632624788
```

Reply:
My point was simply, add a code comment specifying which one is the "interesting" one. It will guard against someone coming in and making it worse at a future date.